src/main/scala/com/exasol/parquetio/reader/RowReadSupport.scala

```diff
@@ -0,0 +1,45 @@
+package com.exasol.parquetio.reader
+
+import com.exasol.parquetio.data.Row
+import com.exasol.parquetio.data.GenericRow
+import com.exasol.parquetio.reader.converter.ParquetRootConverter
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.parquet.hadoop.api.ReadSupport
+import org.apache.parquet.hadoop.api.ReadSupport.ReadContext
+import org.apache.parquet.io.api.RecordMaterializer
+import org.apache.parquet.schema.MessageType
+
+/**
+ * A concrete implementation of [[org.apache.parquet.hadoop.api.ReadSupport]]
+ * that materializes Parquet records into internal [[com.exasol.parquetio.data.Row]]
+ * structure.
+ */
+final class RowReadSupport extends ReadSupport[Row] {
+  override def prepareForRead(
+    conf: Configuration,
+    metadata: java.util.Map[String, String],
+    messageType: MessageType,
+    readContext: ReadContext
+  ): RecordMaterializer[Row] =
+    new RowRecordMaterializer(messageType, readContext)
+
+  override def init(
+    conf: Configuration,
+    metadata: java.util.Map[String, String],
+    messageType: MessageType
+  ): ReadSupport.ReadContext = {
+    val projection = conf.get(ReadSupport.PARQUET_READ_SCHEMA)
+    val requestedSchema = ReadSupport.getSchemaForRead(messageType, projection)
+    new ReadSupport.ReadContext(requestedSchema)
+  }
+
+  private[this] final class RowRecordMaterializer(
+    messageType: MessageType,
+    readContext: ReadContext
+  ) extends RecordMaterializer[Row] {
+    override val getRootConverter: ParquetRootConverter = ParquetRootConverter(messageType)
```

Comment:
```suggestion
    override val rootConverter: ParquetRootConverter = ParquetRootConverter(messageType)
```

Reply:
or is it a function?

Reply:
I think it should be function, I will check and change to def.